api-gateway/src/main/java/com/petclinic/bffapigateway/domainclientlayer/AuthServiceClient.java

```diff
@@ -66,6 +66,16 @@ public Flux<UserDetails> getAllUsers(String jwtToken) {
                 .bodyToFlux(UserDetails.class);
     }
 
+    public Mono<Void> deleteUser(String jwtToken, String userId) {
+        return webClientBuilder.build()
+                .delete()
+                .uri(authServiceUrl + "/users/{userId}", userId)
+                .cookie("Bearer", jwtToken)
+                .retrieve()
+                .onStatus(HttpStatusCode::is4xxClientError, clientResponse -> Mono.error(new GenericHttpException("Error deleting user", HttpStatus.BAD_REQUEST)))
+                .onStatus(HttpStatusCode::is5xxServerError, clientResponse -> Mono.error(new GenericHttpException("Server error", HttpStatus.INTERNAL_SERVER_ERROR)))
+                .bodyToMono(Void.class);
+    }
```

Comment:
you can incorporate this logic:
"private WebClient buildWebClient(String jwtToken) {
    return webClientBuilder.build()
            .cookie("Bearer", jwtToken);
}"

instead of using it in the builder